cpp/include/cudf/types.hpp

```diff
@@ -32,17 +32,20 @@
  * @brief Type declarations for libcudf.
  */
 
+// FIXME: where is this used? TODO: remove if unused.
 namespace bit_mask {
 using bit_mask_t = uint32_t;
 }
 
 // Forward declarations
+/// @cond FORWARD_DECLARATIONS
 namespace rmm {
 class device_buffer;
 namespace mr {
 class device_memory_resource;
 device_memory_resource* get_current_device_resource();
 }  // namespace mr
+/// @endcond
```

Comment:
Using `@cond` to skip this forward declarations by doxygen.

Reply:
Cool trick.

Reply:
This is interesting. When and how would we actually enable `FORWARD_DECLARATIONS`?

Reply:
I would think we would never enable it -- so doxygen skips this part -- if I'm understanding this.
https://www.doxygen.nl/manual/commands.html#cmdcond
It appears the label is optional so we could remove if that is less confusing.